extension/src/popup/helpers/soroban.ts

```diff
@@ -496,11 +508,20 @@ export function getInvocationArgs(
         }
 
         // contractExecutableStellarAsset
-        case 1:
-          return {
+        case 1: {
+          const sacDetails = {
             type: "sac",
             asset: Asset.fromOperation(preimage.fromAsset()).toString(),
-          };
+          } as FnArgsCreateSac;
+
+          if (fn.switch().value === 2) {
```

Comment:
It wasn't super clear to me through the xdr if this was possible for the SAC case but I think there's no reason to not include this line. If SACs will keep using v1 then this conditional won't ever be hit.